src/mlpack/methods/ann/models/yolov3/yolov3_layer.hpp

```diff
@@ -0,0 +1,138 @@
+ *
+ * This layer outputs `gridSize` x `gridSize` x `predictionsPerCell`
+ * bounding boxes.
+ *
+ * For more information, refer to the following paper:
+ *
+ * @code
+ * @article{yolov3,
+ *   title     ={YOLOv3: An Incremental Improvement},
+ *   author    ={Redmon, Joseph and Farhadi, Ali},
+ *   journal   = {arXiv},
+ *   year      ={2018}
+ * }
+ * @endcode
+ *
+ */
+namespace mlpack {
+
+template <typename MatType = arma::mat>
+class YOLOv3Layer : public Layer<MatType>
+{
+ public:
+  YOLOv3Layer() { /* Nothing to do. */ }
+
+  /**
+   * YOLOv3Layer constructor.
+   *
+   * Input dimensions are expected to be 3d, normally from the outputs of a
+   * convolution layer.
+   *
+   * @param imgSize The width and height of input images. Pretrained weights
+       used 416.
+   * @param numAttributes Total number of attributes representing a bounding
+       box. The same as 5 plus the number of classes in the dataset.
+   * @param predictionsPerCell Each YOLO layer predicts `predictionsPerCell`
+       boxes per grid cell. Pretrained weights use 3.
+   * @param anchors Vector of anchor width and heights. Formatted as
+      [w0, h0, w1, h1, ... ]. Each anchors is a [w, h] pair. There must be
+      `predictionsPerCell` anchor pairs.
+   */
+  YOLOv3Layer(const size_t imgSize,
+              const size_t numAttributes,
+              const size_t gridSize,
+              const size_t predictionsPerCell,
+              const std::vector<typename MatType::elem_type> anchors);
+
+  YOLOv3Layer* Clone() const override { return new YOLOv3Layer(*this); }
+
+  // Copy the given YOLOv3Layer.
+  YOLOv3Layer(const YOLOv3Layer& other);
+  // Take ownership of the given YOLOv3Layer.
+  YOLOv3Layer(YOLOv3Layer&& other);
+  // Copy the given YOLOv3Layer.
+  YOLOv3Layer& operator=(const YOLOv3Layer& other);
+  // Take ownership of the given YOLOv3Layer.
+  YOLOv3Layer& operator=(YOLOv3Layer&& other);
+
+  void ComputeOutputDimensions() override;
+
+  /**
+   * NOTE: This will be changed when training is implemented.
+   *
+   * Takes in 3d input and outputs bounding boxes, based on anchors,
+   * input image size and cell position.
+   *
+   * Outputs bounding boxes, whose coordinates are represent by center x and
+   * center y position. If you require x1, y1, x2, y2, you will need to
+   * convert.
+   *
+   * @param input Input data representing outputs of model.
+   * @param output Resulting bounding boxes after being normalized to image.
+   */
+  void Forward(const MatType& input, MatType& output) override;
+
+  /**
+   * NOTE: This will be changed when training is implemented.
+   *
+   * Currently not implemented.
+   */
+  void Backward(const MatType& input,
+                const MatType& output,
+                const MatType& gy,
+                MatType& g) override;
+
+  // Serialize the layer.
+  template<typename Archive>
+  void serialize(Archive& ar, const uint32_t /* version */);
+
+ private:
+  using Type = typename MatType::elem_type;
```

Comment:
A bit longer but I'd call this `ElemType` for consistency with the other layers.